File: backend/srv/src/main/java/com/project/srv/bean/Activite.java
package com.project.srv.bean;
import jakarta.persistence.*;

import java.util.List;

@Entity
public class Activite {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String nom;
    private String image;

    private String lieu;
    private String description;
    private String horaire;
    private double prix;
    @ManyToOne (fetch = FetchType.EAGER)
    private Ville ville;
    @OneToMany(mappedBy = "activite")

    private List<Details> details;
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getLieu() {
        return lieu;
    }
    public String getImage(){return image;}

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public String getDescription() {
        return description;
    }
    public Ville getVille() {
        return ville;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }
    public void setVille(Ville ville) {
        this.ville = ville;
    }
    public void setImage(String image) {
        this.image = image;
    }


}

File: frontend/flutter/travel_mobile/lib/models/activite.dart
class Activity {
  final String nom;
  final String lieu;
  final double prix;
  final String image;
  final String description;
  final String horaire;

  Activity({
    required this.nom,
    required this.lieu,
    required this.prix,
    required this.image,
    required this.description,
    required this.horaire,
  });

  factory Activity.fromJson(Map<String, dynamic> json) {
    return Activity(
      nom: json['nom'],
      lieu: json['lieu'],
      prix: json['prix'],
      image: json['image'],
      description: json['description'],
      horaire: json['horaire'],
    );
  }
}
